ic-os/components/selinux/ic-node/ic-node.te

```diff
@@ -392,6 +386,122 @@ dontaudit ic_canister_sandbox_t ic_canister_sandbox_t : process { signal };
 # Forbid the sandbox from checking certs
 dontaudit ic_canister_sandbox_t cert_t : file { read };
 
+###############################################################################
+# Compiler sandbox
+
+role system_r types ic_compiler_sandbox_t;
+
+require {
+    type ld_so_cache_t, ld_so_t, lib_t, root_t, usr_t, cpu_online_t, proc_t, sysfs_t, null_device_t;
+    attribute unconfined_domain_type;
+}
+
+allow ic_compiler_sandbox_t etc_t:dir { getattr ioctl lock open read search };
+allow ic_compiler_sandbox_t ld_so_cache_t:file { getattr ioctl lock map open read };
+allow ic_compiler_sandbox_t ld_so_t:file { execute getattr map open read };
+allow ic_compiler_sandbox_t ld_so_t:lnk_file { getattr read };
+allow ic_compiler_sandbox_t lib_t:dir { getattr lock open read search };
+allow ic_compiler_sandbox_t lib_t:file { execute getattr map open read };
+allow ic_compiler_sandbox_t lib_t:lnk_file { getattr read };
+allow ic_compiler_sandbox_t root_t:dir { getattr read search };
+allow ic_compiler_sandbox_t root_t:lnk_file { getattr read };
+allow ic_compiler_sandbox_t unconfined_domain_type:association recvfrom;
+allow ic_compiler_sandbox_t usr_t:dir { getattr open search };
+allow ic_compiler_sandbox_t ic_compiler_sandbox_exec_t:file { entrypoint execute map read };
+allow ic_compiler_sandbox_t ic_compiler_sandbox_t:process { fork getsched };
+
+# Allow to turn anonymous memory executable. This is obviously not desirable
+# from security POV, but it is better than having `execmem` permissions on the
+# canister sandbox because here we are only compiling untrusted user Wasm code -
+# not executing it.
+# This is reqired because the wasmtime compilation function writes the code to
+# an executable region before dumping it to a file. That could be changed, but
+# would require upstreaming a change to wasmtime.
+allow ic_compiler_sandbox_t self : process { execmem };
```

Comment:
The compiler sandbox is a little more restricted than the canister sandbox. This is the only case where the compiler sandbox has more permissions than the canister sandbox.